extensions/EntityExtension/src/main/kotlin/com/typewritermc/entity/entries/activity/PlayerCloseByActivity.kt

```diff
@@ -86,7 +100,24 @@ class PlayerCloseByActivity(
                 .update(player.location.toProperty())
         }
 
-        val isActive = trackers.any { (_, tracker) -> tracker.isActive(maxIdleDuration) }
+        val trackerActive = trackers.any { (_, tracker) -> tracker.isActive(maxIdleDuration) }
+
+        isActive = when {
+            isActive -> trackerActive
+            trackerActive -> if (activationRange.isPresent) {
+                val activationThresholdSq = activationRange.get() * activationRange.get()
+                trackers.any { (_, tracker) ->
+                    tracker.isActive(maxIdleDuration) &&
+                            (tracker.location.distanceSqrt(currentPosition)
+                                ?: Double.MAX_VALUE) < activationThresholdSq
+                }
+            } else {
+                true
+            }
+
+            else -> false
+        }
+
```

Comment:
I think this would be better shouted in the location tracker class instead of here.